code refactoring

deepserket opened the pull request:
I have just a question, why all the check functions are in the `Account` class except `check_length`?
Since in the utils module is marked as "_useful for external consumption_" I haven't moved the `check_length` function

deepserket committed:
code refactoring

deepserket committed:
code refactoring

clamytoe commented:
To be honest, not sure. I think it made sense at the time. Should be in the Account class though. Go ahead and move it. Once you do I’ll run the tests and if all is good I’ll go ahead and merge it. Your code looks good so far!

deepserket committed:
moved check_length into the Account class

deepserket committed:
moved check_length into the Account class, 

Also changed the order of the checks, since the "not_in_dict" function is the slowest it's better to call it after doing the other checks.
Eg. imagine having spent 10 minutes checking the dictionaries and then the password is not validated just because it contains the username, or because it is old, etc ...

deepserket committed:
Update passwd_validate.py

clamytoe committed:
Fixed PR

Method _check_length() was missing self.
